qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.upgrades;
+
+import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.xcontent.json.JsonXContent;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.equalTo;
+
+public class FieldCapsIT extends AbstractRollingTestCase {
+    public void testFieldCaps() throws Exception {
+        final String redMapping = """
+             "properties": {
+               "red_field": { "type": "keyword" },
+               "yellow_field": { "type": "integer" },
+               "timestamp": {"type": "date"}
+             }
+            """;
+        final String greenMapping = """
+             "properties": {
+               "green_field": { "type": "keyword" },
+               "yellow_field": { "type": "long" },
```

Comment:
is it on purpose that this filed is long here but integer above?

Reply:
Yes, that was on purpose. I added a new field that has the same type in both indices.